### PR TITLE
Improve test coverage for auto_executable_network.cpp

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *Improve test coverage for auto_executable_network.cpp*

### Tickets:
 - *ticket-99028*

### CODE REVIEW

File: src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp [hunk 1]
```diff
@@ -8,6 +8,9 @@
 #include "multi/multi_remote_blob_tests.hpp"
 #include "multi/multi_remote_blob_multidevice_test.hpp"
 #include "common_test_utils/test_constants.hpp"
+#include <openvino/runtime/auto/properties.hpp>
+
+using MultiDevice_RemoteBlobAndBindTest = MultiDevice_Test;
```
Comment (songbell) [inline src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp:13]: let's use MultiDevice_Test only, bind can be tested through instantiating params with config DEVICE_BIND_BUFFER(YES)

**Comment by songbell** [comment]
also,  better to modify the title a little bit, looks like there are some other cases not involved yet(maybe in another PR?)

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_correctness.cpp [hunk 2]
```diff
@@ -45,6 +45,13 @@ void OVInferConsistencyTest::SetUp() {
     for (auto&& item : _deviceConfigs) {
         ModelContext modelContext;
         modelContext._model = core->compile_model(function, item.first, item.second);
+        if (_inferReqNumPerModel == 0) {
+            try {
+                _inferReqNumPerModel =  modelContext._model.get_property(ov::optimal_number_of_infer_requests);
+            } catch (...) {
+                throw("cannot deduce infer request number");
+            }
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_correctness.cpp:53]: why add this? what error do we encounter?

**Comment by songbell** [comment]
also, maybe we can modify the title a little bit (like phase I or something other), as I can see there are some other cases not been added in this PR(maybe in another PR)?

**Review by wangleis — APPROVED** [review]

File: src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_infer_request/perf_counters.cpp [hunk 3]
```diff
@@ -34,11 +34,18 @@ auto configs = []() {
 };
 
 auto Multiconfigs = []() {
-    return std::vector<ov::AnyMap>{{ov::device::priorities(CommonTestUtils::DEVICE_GPU)}};
+    return std::vector<ov::AnyMap>{{ov::device::priorities(CommonTestUtils::DEVICE_GPU)},
+                                   {ov::device::priorities(CommonTestUtils::DEVICE_GPU, CommonTestUtils::DEVICE_CPU),
```
Comment (songbell) [inline src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_infer_request/perf_counters.cpp:38]: generally, we are using macro ENABLE_INTEL_CPU to make sure tests need CPU will be scheduled to suitable hardware nodes.

File: src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp [hunk 4]
```diff
@@ -8,20 +8,29 @@
...
+        {{"GPU.0"}, true, {}},  // GPU.0 via MULTI,
+        {{GPU}, true, {ov::intel_auto::device_bind_buffer(true)}},      // GPU via MULTI,
+        {{"GPU.0"}, true, {ov::intel_auto::device_bind_buffer(true)}},  // GPU.0 via MULTI,
 #ifdef ENABLE_INTEL_CPU
-        {{GPU, CPU}, true},  // GPU+CPU
-        {{CPU, GPU}, true},  // CPU+GPU
+        {{GPU, CPU}, true, {}},  // GPU+CPU
+        {{CPU, GPU}, true, {}},  // CPU+GPU
+        {{GPU, CPU}, true, {ov::intel_auto::device_bind_buffer(true)}},  // GPU+CPU
+        //{{CPU, GPU}, true, {ov::intel_auto::device_bind_buffer(true)}},  // CPU+GPU hang
```
Comment (songbell) [inline src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp:25]: why there is hang?

Comment (wgzintel) [reply src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp:25]: CPU+GPU hangs when infer request waits for result. it Hangs in src/tests/functional/plugin/shared/include/multi/multi_remote_blob_tests.hpp line 36

Comment (songbell) [reply src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp:25]: we have to understand the problem and resolve it

File: src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_infer_request/perf_counters.cpp (hunk 3 above)
Comment (wgzintel) [reply src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_infer_request/perf_counters.cpp:39]: updated

File: src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp (hunk 4 above)
Comment (wgzintel) [reply src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp:25]: disccuss offline

File: src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp [hunk 5]
```diff
@@ -50,29 +63,86 @@ TEST_P(MultiDevice_Test, cannotInferRemoteBlobIfNotInitializedForDevice) {
...
+INSTANTIATE_TEST_SUITE_P(smoke_Multi_RemoteBlobInitializedWithoutGPU,
                          MultiDevice_Test,
                          ::testing::ValuesIn(device_names_and_support_for_remote_blobs2()),
                          MultiDevice_Test::getTestCaseName);
 
+auto device_names_and_support_for_remote_blobs3 = []() {
+    return std::vector<DevicesNamseAndProperties>{
+#ifdef ENABLE_INTEL_CPU
+        {{CPU}, {ov::intel_auto::device_bind_buffer(true)}},  // stand-alone CPU via MULTI (no GPU), no OCL context
+#endif
```
Comment (songbell) [inline src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp:127]: if this is only cpu only, move to cpu func tests, or use gpu only, or just delete this test

File: src/tests/functional/plugin/shared/include/multi/multi_remote_blob_multidevice_test.hpp [hunk 6]
```diff
@@ -20,7 +20,8 @@ TEST_P(MultiDeviceMultipleGPU_Test, canCreateRemoteTensorThenInferWithAffinity)
     auto function = p.build();
     ov::CompiledModel exec_net;
     try {
-        exec_net = ie.compile_model(function, device_names, ov::hint::allow_auto_batching(false));
+        exec_net = ie.compile_model(function, device_names, {ov::hint::allow_auto_batching(false),
+            ov::hint::performance_mode(ov::hint::PerformanceMode::UNDEFINED)});
```
Comment (songbell) [inline src/tests/functional/plugin/shared/include/multi/multi_remote_blob_multidevice_test.hpp:24]: revert this line, no need in current master

**Review by songbell — APPROVED** [review]

File: src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp (hunk 5 above)
Comment (wgzintel) [reply src/tests/functional/plugin/gpu/shared_tests_instances/multi/gpu_remote_blob_tests.cpp]: Updated, replace CPU with GPU.

File: src/tests/functional/plugin/shared/include/multi/multi_remote_blob_multidevice_test.hpp (hunk 6 above)
Comment (wgzintel) [reply src/tests/functional/plugin/shared/include/multi/multi_remote_blob_multidevice_test.hpp]: Updated, revert line.

**Review by wangleis — APPROVED** [review]

**Pull request merged**